packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,420 @@
+import { blockFromRpc } from '@ethereumjs/block/dist/from-rpc'
+import { RLP } from '@ethereumjs/rlp'
+import { Trie } from '@ethereumjs/trie'
+import {
+  Account,
+  arrToBufArr,
+  bigIntToHex,
+  bufferToHex,
+  intToHex,
+  isHexPrefixed,
+  toBuffer,
+} from '@ethereumjs/util'
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { debug } from 'debug'
+import { hexToBytes } from 'ethereum-cryptography/utils'
+
+import { Cache } from './cache'
+
+import { BaseStateManager } from '.'
+
+import type { Proof, StateManager } from '.'
+import type { StorageDump } from './interface'
+import type { StorageProof } from './stateManager'
+import type { Common } from '@ethereumjs/common'
+import type { Address, PrefixedHexString } from '@ethereumjs/util'
+
+const log = debug('statemanager')
+export interface EthersStateManagerOpts {
+  provider: string | JsonRpcProvider
+  blockTag?: bigint | 'latest' | 'earliest'
```

Comment:
rather than having block number/tag as part of constructor opts, can we have a `setBlock` fn that does the job, in this way it can be used recurringly

Reply:
I think I'll keep it as optional so people can set it once and forget it if they are only running one call and then add a second initalization method to allow you to set the blockTag (which in turn will reset the local state/storage cache.